BizHawk.Emulation.Cores/Consoles/Nintendo/NES/NES.Core.cs

```diff
@@ -736,21 +774,63 @@ void write_joyport(byte value)
 			var si = new StrobeInfo(latched4016, value);
 			ControllerDeck.Strobe(si, _controller);
 			latched4016 = value;
+			new_strobe = (value & 1) > 0;
+			if (current_strobe && !new_strobe)

```

Comment:
From what I remember, the classes in the ControllerDeck already handle all aspects of strobing and bitshifting, and they can be savestated at any point in the process without issue.  Why is all of that logic being duplicated here?

Reply:
It does, but it wasn't obvious to me how to get all the menus working to pick controllers and I wanted to have something put together for testing. I'm hoping over the next few days I'll have some time to get the NES menu and ppu viewer up and running. Then that redundant stuff will go away.